backend/commons/src/main/java/org/sonarsource/sonarlint/core/commons/util/git/WinGitUtils.java

```diff
@@ -30,19 +32,28 @@ private WinGitUtils() {
   }
 
   public static Optional<String> locateGitOnWindows() {
+    return locateGitOnWindows(WinGitUtils::callWhereTool);
+  }
+
+  static Optional<String> locateGitOnWindows(Supplier<ProcessWrapperFactory.ProcessExecutionResult> gitExeLocationSupplier) {
     // Windows will search current directory in addition to the PATH variable, which is unsecure.
     // To avoid it we use where.exe to find git binary only in PATH.
-    LOG.debug("Looking for git command in the PATH using where.exe (Windows)");
-    var result = new ProcessWrapperFactory()
-      .create(null, "C:\\Windows\\System32\\where.exe", "$PATH:git.exe")
-      .execute();
+    var result = gitExeLocationSupplier.get();
 
-    if (result.exitCode() == 0) {
-      var out = result.output().trim();
+    var output = result.output();
+    if (result.exitCode() == 0 && output.contains("git.exe")) {
+      var out = Arrays.stream(output.split(System.lineSeparator())).map(String::trim).findFirst();
```

Comment:
It might not happen often, but as we randomly pick the first found Git, we might end up using an older version than what we need, while another Git would have suited better. But the minimal version we require is 6y old, so maybe it wouldn't be a common issue

Reply:
Concern is fair, but I suggest not to write the version picking algorithm now.
Ideally yes, we should call the `git --version` here and make sure it's good for us. If not, check another version and fallback to JGit if none is OK.
I suggest to proceed this way now.